mteb/evaluation/evaluators/PairClassificationEvaluator.py

```diff
@@ -81,22 +110,17 @@ def compute_metrics(
         if "batch_size" not in encode_kwargs:
             encode_kwargs["batch_size"] = 32
 
-        sentences = list(set(self.sentences1 + self.sentences2))
-
-        total_sents = len(self.sentences1) + len(self.sentences2)
-        n_duplicates = total_sents - len(sentences)
-        if n_duplicates:
-            logger.warning(
-                f"Found {n_duplicates}/{total_sents} duplicates in the input data. Only encoding unique sentences."
-            )
-        embeddings = model.encode(
-            sentences,
+        all_sentences = self.sentences1 + self.sentences2
+        len_sentences1 = len(self.sentences1)
+        embeddings = self._encode_unique_texts(
+            all_sentences,
+            model,
             task_name=self.task_name,
+            prompt_type=PromptType.query,
```

Comment:
PairClassification shouldn't have prompt_type

Reply:
changed it!

Reply:
I think that it should be removed totally from evaluator. Also can you run some tasks to check scores?

Reply:
Results for SprintDuplicateQuestions match the ones in [the results repo](https://github.com/embeddings-benchmark/results/blob/main/results/sentence-transformers__all-MiniLM-L6-v2/8b3219a92973c328a8e22fadcfa821b5dc75636a/SprintDuplicateQuestions.json) for sentence-transformers__all-MiniLM-L6-v2. 

```
{'default': {'similarity_accuracy': 0.9978514851485148, 'similarity_accuracy_threshold': 0.7130638957023621, 'similarity_f1': 0.890126582278481, 'similarity_f1_threshold': 0.7130638957023621, 'similarity_precision': 0.9015384615384615, 'similarity_recall': 0.879, 'similarity_ap': 0.9455114300045775, 'cosine_accuracy': 0.9978514851485148, 'cosine_accuracy_threshold': 0.7130638957023621, 'cosine_f1': 0.890126582278481, 'cosine_f1_threshold': 0.7130638957023621, 'cosine_precision': 0.9015384615384615, 'cosine_recall': 0.879, 'cosine_ap': 0.9455114300045775, 'manhattan_accuracy': 0.9978415841584158, 'manhattan_accuracy_threshold': 11.838340759277344, 'manhattan_f1': 0.8898989898989899, 'manhattan_f1_threshold': 11.838340759277344, 'manhattan_precision': 0.8989795918367347, 'manhattan_recall': 0.881, 'manhattan_ap': 0.9453938408856842, 'euclidean_accuracy': 0.9978514851485148, 'euclidean_accuracy_threshold': 0.7575434446334839, 'euclidean_f1': 0.890126582278481, 'euclidean_f1_threshold': 0.7575434446334839, 'euclidean_precision': 0.9015384615384615, 'euclidean_recall': 0.879, 'euclidean_ap': 0.9455114300045776, 'dot_accuracy': 0.9978514851485148, 'dot_accuracy_threshold': 0.7130638957023621, 'dot_f1': 0.890126582278481, 'dot_f1_threshold': 0.7130638957023621, 'dot_precision': 0.9015384615384615, 'dot_recall': 0.879, 'dot_ap': 0.9455114300045775, 'max_accuracy': 0.9978514851485148, 'max_f1': 0.890126582278481, 'max_precision': 0.9015384615384615, 'max_recall': 0.881, 'max_ap': 0.9455114300045776, 'main_score': 0.9455114300045776}}
```